### ISSUE TITLE
Running with Superset and SQLLab - unable to list tables.

### ISSUE THREAD

**Issue by darylerwin** [mxmzdlv/pybigquery#10]
When running SqlLab and SqlEditor it tries to fill in the Database, Schema then the tables for the Schema. It properly populates the 2, but then when it tries to read/parse the tables for the schema it errors out.
Error While Fetching Table List is the superset error.

(How can one debug something like this? I tried adding logging.debug('blah blah') but never got any output)

It appears the call to 
GET https://www.googleapis.com/bigquery/v2/projects/projectId/datasets/datasetId/tables
is the problem.

The python dump for the pybigquery is as follows:
```
2018-02-10 19:09:21,610:INFO:root:Database.get_sqla_engine(). Masked URL: bigquery://production-bigdata/
2018-02-10 19:09:21,620:DEBUG:urllib3.util.retry:Converted retries value: 3 -> Retry(total=3, connect=None, read=None, redirect=None, status=None)
2018-02-10 19:09:21,622:DEBUG:google.auth.transport.requests:Making request: POST https://accounts.google.com/o/oauth2/token
2018-02-10 19:09:21,627:DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): accounts.google.com
2018-02-10 19:09:21,696:DEBUG:urllib3.connectionpool:https://accounts.google.com:443 "POST /o/oauth2/token HTTP/1.1" 200 None
2018-02-10 19:09:21,700:DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): www.googleapis.com
2018-02-10 19:09:21,804:DEBUG:urllib3.connectionpool:https://www.googleapis.com:443 "GET /bigquery/v2/projects/production-bigdata/datasets HTTP/1.1" 200 None
2018-02-10 19:09:21,805:ERROR:root:dataset must be a Dataset or a DatasetReference
Traceback (most recent call last):
  File "/home/derwin/venv/local/lib/python2.7/site-packages/superset/views/base.py", line 62, in wraps
    return f(self, *args, **kwargs)
  File "/home/derwin/venv/local/lib/python2.7/site-packages/flask_appbuilder/security/decorators.py", line 52, in wraps
    return f(self, *args, **kwargs)
  File "/home/derwin/venv/local/lib/python2.7/site-packages/superset/views/core.py", line 1335, in tables
    database, database.all_table_names(schema), schema)
  File "/home/derwin/venv/local/lib/python2.7/site-packages/superset/models/core.py", line 724, in all_table_names
    self.db_engine_spec.get_table_names(schema, self.inspector))
  File "/home/derwin/venv/local/lib/python2.7/site-packages/superset/db_engine_specs.py", line 225, in get_table_names
    
  File "/home/derwin/venv/local/lib/python2.7/site-packages/sqlalchemy/engine/reflection.py", line 201, in get_table_names
    self.bind, schema, info_cache=self.info_cache)
  File "/home/derwin/venv/local/lib/python2.7/site-packages/pybigquery/sqlalchemy_bigquery.py", line 192, in get_table_names
    tables = connection.connection._client.list_dataset_tables(d)
  File "/home/derwin/venv/local/lib/python2.7/site-packages/google/cloud/bigquery/client.py", line 459, in list_dataset_tables
    return self.list_tables(*args, **kwargs)
  File "/home/derwin/venv/local/lib/python2.7/site-packages/google/cloud/bigquery/client.py", line 438, in list_tables
    raise TypeError('dataset must be a Dataset or a DatasetReference')
TypeError: dataset must be a Dataset or a DatasetReference
```

**Comment by darylerwin** [comment]
Think I found it .. atleast my test script is working (will test on superset tomorrow).
```
#        tables = connection.connection._client.list_tables(d)
          tables = connection.connection._client.list_tables(connection.connection._client.dataset(d.dataset_id))

```

My test code:
```
from sqlalchemy import *
from sqlalchemy.engine import create_engine
from sqlalchemy.schema import *
from pprint import pprint

engine = create_engine('bigquery://production-bigdata',echo='debug')
insp = inspect(engine)

pprint (insp.get_schema_names())
pprint (insp.get_table_names())
pprint ('===============DEBUG============')
pprint (insp.get_table_names('somedataset'))
```

**Comment by mxmzdlv** [comment]
Thanks for the report! I've just pushed a fix, can you please upgrade pybigquery to 0.2.7 and check if it is all good now? There was a compatibility issue with the latest google-cloud-bigquery client.

**Comment by darylerwin** [comment]
Yes .. you have corrected the issue with the table list with the update. Thanks!
I have another error that I will post in a new thread...
